src/span.h

```diff
@@ -42,18 +67,59 @@ class Span
     /** Default assignment operator. */
     Span& operator=(const Span& other) noexcept = default;
```

Comment:
I'm pretty sure that these default declarations are unnecessary, since they'll be generated by default anyway.

The only place I'm aware of needing to declare default ctors/assignment operators is when you're using a unique_ptr for an incomplete type, and you need to declare them in the header file and define them (as default) in the implementation file (eg when using a unique_ptr for a pimpl).

They may be other situations where declaring default ctors/assiment operators is necessary, but I don't think this is one of them.

Reply:
> I'm pretty sure that these default declarations are unnecessary, since they'll be generated by default anyway.

Built, tested, and fuzzed this PR and #19326 with the default copy constructor and assignment operator in `span.h` both removed and yes, AFAICT it seems fine.

Reply:
I believe the default declarations aren't actually a no-op here and may be considered harmful as default move ctor/assignment is disabled when a copy/assignment oper are provided.

See [here under  Implicitly-declared move constructor for reference](https://en.cppreference.com/w/cpp/language/move_constructor). From a quick local test, it seems that even declaring a ```= default``` constructor/assignment operator is enough to disable the implicit moves. Please correct me if I'm wrong on that, though!

So as a side-effect, removing these (thus implicitly creating the move ctor) is enough to eliminate the ```lifetimebound``` false-positives reported [here](https://github.com/bitcoin/bitcoin/pull/19367#discussion_r445025361).

+1 for removing them unless there was a reason for disabling those moves.

Reply:
> it seems that even declaring a = default constructor/assignment operator is enough to disable the implicit moves.

Fascinating! I didn't know that.

> +1 for removing them unless there was a reason for disabling those moves.

If this is the case, presumably it'd be better to explicitly delete those move ctor/assigments than rely on this behavior.

Reply:
> > it seems that even declaring a = default constructor/assignment operator is enough to disable the implicit moves.
> 
> Fascinating! I didn't know that.

Please double-check me on this! I'm assuming that based on my testing with lifetimebound, but I haven't managed to track down a definitive answer.